src/Modules/Module.php

```diff
@@ -14,29 +15,26 @@ abstract class Module implements \Webleit\ZohoCrmApi\Contracts\Module
 {
     use HasInflector;
 
-    const RESPONSE_OPTION_PAGINATION_ONLY = 2;
-
-    const TRIGGER_WORKFLOW = 'workflow';
-    const TRIGGER_BLUEPRINT = 'blueprint';
-    const TRIGGER_APPROVAL = 'approval';
-
     /**
      * @var Client
      */
     protected $client;
 
-    protected $pagination;
-
     public function __construct(Client $client)
     {
         $this->client = $client;
     }
 
-    public function getList(array $params = []): RecordCollection
+    public function getList(array $params = [], array $headers = []): RecordCollection
     {
-        $list = $this->client->getList($this->getUrl(), $params);
+        $list = $this->client->getList($this->getUrl(), $params, $headers);
+
+        $data = $list[$this->getResourceKey()] ?? null;
+        if ($data === null) {
+            throw new InvalidResourceKey(json_encode($list));
```

Comment:
This will throw if the If Modified Since header is set and there are no results. The reasoning is that Zoho returns a 304 with no body